pkg/operator/enable_pacemaker_v1.go

```diff
@@ -0,0 +1,152 @@
+package operator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/trento-project/workbench/internal/systemd"
+)
+
+const (
+	EnablePacemakerOperatorName = "enablepacemaker"
+	pacemakerServiceName        = "pacemaker.service"
+)
+
+type pacemakerEnablementDiffOutput struct {
+	Enabled bool `json:"enabled"`
+}
+
+type EnablePacemakerOption Option[EnablePacemaker]
+
+// EnablePacemaker operator enables Pacemaker systemd unit.
+//
+// # Execution Phases
+//
+// - PLAN:
+//   The operator connects to systemd and determines if the Pacemaker service is enabled.
+//   The operation is skipped if the Pacemaker service is already enabled.
+//
+// - COMMIT:
+//   It enables the Pacemaker systemd unit.
+//
+// - VERIFY:
+//   The operator checks if the Pacemaker service is enabled after the commit phase.
+//
+// - ROLLBACK:
+//   If an error occurs during the COMMIT or VERIFY phase, the Pacemaker service is disabled back again.
+
+type EnablePacemaker struct {
+	baseOperator
+	systemdLoader    systemd.SystemdLoader
+	systemdConnector systemd.Systemd
+}
+
+func WithCustomEnablePacemakerSystemdLoader(systemdLoader systemd.SystemdLoader) EnablePacemakerOption {
+	return func(o *EnablePacemaker) {
+		o.systemdLoader = systemdLoader
+	}
+}
+
+func NewEnablePacemaker(
+	arguments OperatorArguments,
+	operationID string,
+	options OperatorOptions[EnablePacemaker],
+) *Executor {
+	enablePacemaker := &EnablePacemaker{
+		baseOperator:  newBaseOperator(operationID, arguments, options.BaseOperatorOptions...),
+		systemdLoader: systemd.NewDefaultSystemdLoader(),
+	}
+
+	for _, opt := range options.OperatorOptions {
+		opt(enablePacemaker)
+	}
+
+	return &Executor{
+		phaser:      enablePacemaker,
+		operationID: operationID,
+	}
+}
+
+func (s *EnablePacemaker) plan(ctx context.Context) (bool, error) {
+	if s.systemdConnector == nil {
```

Comment:
I wonder if it is worthy the `systemLoader` interface just to the test `NewSystemd` call.
I mean, it is just one if statement.

Either way, with this, we don't need the `if s.systemdConnector == nil` comparison, right?
It will always we `nil` as we don't initialize it never before. Or am I missing something else?

PD:
You knew this was coming hehe

Reply:
No strong opinions.
Simply I wouldn't put it hehe
I find it an acceptable trade-off. The posiblities of having a bug there are almost 0.
It is fine though. Everyone has its own style :+1: